circleguard/comparer.py

```diff
@@ -144,7 +146,35 @@ def _compare_two_replays(replay1, replay2):
         if (Mod.HR in replay1.mods) ^ (Mod.HR in replay2.mods):
             xy1[:, 1] = 384 - xy1[:, 1]
 
-        return Comparer._compute_data_similarity(xy1, xy2)
+        (play_sequence_1, play_sequence_2) = Comparer.convert_to_row_major_form(xy1, xy2)
+
+        NUM_CHUNKS = 20
+
+        # Sectioned into 20 chunks, used to ignore outliers (eg. long replay with breaks is copied, and the cheater cursordances during the break)
+        horizontal_length = play_sequence_1.shape[1] - play_sequence_1.shape[1] % NUM_CHUNKS
+        play_sequence_1_sections = np.hsplit(play_sequence_1[:,:horizontal_length], NUM_CHUNKS)
+        play_sequence_2_sections = np.hsplit(play_sequence_2[:,:horizontal_length], NUM_CHUNKS)
+        correlations = []
+        for (play_sequence_1, play_sequence_2) in zip(play_sequence_1_sections, play_sequence_2_sections):
+            cross_correlation_matrix = Comparer.find_cross_correlation(play_sequence_1, play_sequence_2)
+            #Pick the lag with the maximum correlation, this likely in most cases is 0 lag
+            max_correlation = max(cross_correlation_matrix.reshape(cross_correlation_matrix.size))
+            correlations.append(max_correlation)
+        average_distance = Comparer._compute_data_similarity(xy1, xy2)
+        # Out of all the sections, we should have 20 sections, we pick the one with the median correlation, so we throw away any outliers
+        median_correlation = np.median(np.array(sorted(correlations)))
+        return (average_distance, median_correlation)
+
+    @staticmethod
+    def convert_to_row_major_form(x, y):
+        return (np.array(list(zip(*x))), np.array(list(zip(*y))))
+
+    @staticmethod
+    def find_cross_correlation(x, y):
+        # The normalization is makes the similarity detector robust to translations
+        x = (x - np.mean(x)) / (np.std(x) * x.size)
```

Comment:
you should probably do this normalization after the cross correlation, as this normalizes all data when doing it after only normalizes 1 number

Reply:
Hmm, still unsure what you mean, maybe post some code, or make the changes you want to this pr?

Reply:
something like
```
x -= np.mean(x)
y -= np.mean(y)
norm = np.std(x) * np.std(y) * x.size
return signal.correlate(x, y) / norm
```